clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/VenicePushJob.java

```diff
@@ -1346,14 +1355,6 @@ protected static boolean shouldBuildZstdCompressionDictionary(
       PushJobSetting pushJobSetting,
       StoreSetting storeSetting,
       boolean inputFileHasRecords) {
-    if (pushJobSetting.isSourceKafka) {
```

Comment:
Sure, I can revert this.
Currently, the logic about dict build is a little confusing, and I hope we could refactor it in the future to avoid confusion.

Reply:
I will take a shot at this sometime over the next couple of weeks. Thanks for the feedback.